x/stakeibc/keeper/unbonding_records_sweep_unbonded_tokens_test.go

```diff
@@ -0,0 +1,204 @@
+		epochUnbonding.HostZoneUnbondings = []*recordtypes.HostZoneUnbonding{
+			epochUnbonding.HostZoneUnbondings[0],
+		}
+		s.App.RecordsKeeper.SetEpochUnbondingRecord(s.Ctx(), epochUnbonding)
+	}
+	success, successful_sweeps, sweep_amounts, failed_sweeps := s.App.StakeibcKeeper.SweepAllUnbondedTokens(s.Ctx())
+	s.Require().Equal(success, false, "sweep all tokens failed if osmo missing")
+	s.Require().Equal(len(successful_sweeps), 1, "sweep all tokens succeeds for 1 host zone")
+	s.Require().Equal(successful_sweeps[0], "GAIA", "sweep all tokens succeeds for gaia")
+	s.Require().Equal(len(sweep_amounts), 1, "sweep all tokens succeeds for 1 host zone")
+	s.Require().Equal(len(failed_sweeps), 1, "sweep all tokens fails for 1 host zone")
+	s.Require().Equal(failed_sweeps[0], "OSMO", "sweep all tokens fails for osmo")
+	s.Require().Equal(sweep_amounts, []int64{2_000_000}, "correct amount of tokens swept for each host zone")
+}
+
+func (s *KeeperTestSuite) TestSweepUnbondedTokens_RedemptionAccountMissing() {
+	tc := s.SetupSweepUnbondedTokens()
+	_ = tc
+	hostZone, _ := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), "GAIA")
+	hostZone.RedemptionAccount = nil
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+	success, successful_sweeps, sweep_amounts, failed_sweeps := s.App.StakeibcKeeper.SweepAllUnbondedTokens(s.Ctx())
+	s.Require().Equal(success, false, "sweep all tokens failed if osmo missing")
+	s.Require().Equal(len(successful_sweeps), 1, "sweep all tokens succeeds for 1 host zone")
+	s.Require().Equal(successful_sweeps[0], "OSMO", "sweep all tokens succeeds for osmo")
+	s.Require().Equal(len(sweep_amounts), 1, "sweep all tokens succeeds for 1 host zone")
+	s.Require().Equal(len(failed_sweeps), 1, "sweep all tokens fails for 1 host zone")
+	s.Require().Equal(failed_sweeps[0], "GAIA", "sweep all tokens fails for gaia")
+	s.Require().Equal(sweep_amounts, []int64{3_000_000}, "correct amount of tokens swept for each host zone")
+}
+
+func (s *KeeperTestSuite) TestSweepUnbondedTokens_DelegationAccountAddressMissing() {
+	tc := s.SetupSweepUnbondedTokens()
+	_ = tc
+	hostZone, _ := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), "OSMO")
+	hostZone.DelegationAccount.Address = ""
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+	success, successful_sweeps, sweep_amounts, failed_sweeps := s.App.StakeibcKeeper.SweepAllUnbondedTokens(s.Ctx())
+	s.Require().Equal(success, false, "sweep all tokens failed if gaia missing")
+	s.Require().Equal(len(successful_sweeps), 1, "sweep all tokens succeeds for 1 host zone")
+	s.Require().Equal(successful_sweeps[0], "GAIA", "sweep all tokens succeeds for gaia")
+	s.Require().Equal(len(sweep_amounts), 1, "sweep all tokens succeeds for 1 host zone")
+	s.Require().Equal(len(failed_sweeps), 1, "sweep all tokens fails for 1 host zone")
+	s.Require().Equal(failed_sweeps[0], "OSMO", "sweep all tokens fails for osmo")
+	s.Require().Equal(sweep_amounts, []int64{2_000_000}, "correct amount of tokens swept for each host zone")
```

Comment:
might make sense to throw these in a function `checkStateAfterSweep` that takes an argument the dictates which hz should fail. Would make it a little easier to keep these in sync

Reply:
I believe the checks are slightly different so didn't do this for now, although we can always add it in if desired!

Reply:
Sorry, to be clear, I meant parameterize them. Could be something like:
```
checkStateAfterSweep(gaiaSucceeded bool, osmoSucceed bool)
```
But I think we should just address this later once we refactor the tests to the `ForHostZone` function